src/index.mjs

```diff
@@ -190,13 +211,65 @@ async function processCondaEnvironment(filePath) {
     }
 }
 
+// Fetch modified lines in the PR
+async function getModifiedLinesFromCommitDiff(filePath) {
+    try {
+        const { stdout, stderr } = await execPromise(`git diff HEAD^ HEAD -- ${filePath}`);
+        if (stderr) {
+            throw new Error(`Error fetching commit diff: ${stderr}`);
+        }
+
+        const patchLines = stdout.split('\n');
+        const modifiedLines = [];
+
+        let lineNumber = 0;
+
+        // Parse the diff to find the modified lines (based on Git diff format)
+        for (const line of patchLines) {
+            if (line.startsWith('@@')) {
+                const match = /@@ -\d+,\d+ \+(\d+),/.exec(line);
+                lineNumber = match ? parseInt(match[1], 10) : lineNumber;
+            } else if (line.startsWith('+') && !line.startsWith('+++')) {
+                // Track the modified line numbers
+                modifiedLines.push(lineNumber);
+                lineNumber++;
+            } else if (!line.startsWith('-')) {
+                // Increment the line number for non-removed lines
+                lineNumber++;
+            }
+        }
+
+        return modifiedLines;
+    } catch (error) {
+        core.setFailed(`Error getting modified lines from commit diff: ${error.message}`);
+        return [];
+    }
+}
+
 async function run() {
     const ecosystem = core.getInput('package-ecosystem', { required: true });
+    const annotateModifiedOnly = core.getInput('annotate-modified-only') === 'true';
+    let modifiedLines = [];
```

Comment:
if you set this to an array of the file line count you can simplify below

```
        if (modifiedLines.length > 0) {
            await processLines('requirements.txt', modifiedLines, 'pip');
        } else {
            await processPipRequirements('requirements.txt');
        }
```

to just:
```
        await processLines('requirements.txt', modifiedLines, 'pip');
        
```